javascript/script.js

```diff
@@ -0,0 +1,185 @@
+
+  function lowerCaseOption() {
+    const option = confirm(
+      "Press OK if you want your password to have lowercase characters"
+    );
+    return option;
+  }
+
+  //                       STEP 4 :
+  //  function for confirm bar for whether the user wants uppercase characters in password
+
+  function upperCaseOption() {
+    const option = confirm(
+      "Press OK if you want your password to have uppercase characters"
+    );
+    return option;
+  }
+
+  //                       STEP 5 :
+  //  function for confirm bar for whether the user wants numbers characters in password
+
+  function numericOption() {
+    const option = confirm(
+      "Press OK if you want your password to have Numbers"
+    );
+    return option;
+  }
+
+  //                       STEP 6 :
+  //  function for confirm bar for whether the user wants special characters in password
+  function specialCharacterOption() {
+    const option = confirm(
+      "Press OK if you want your password to have special characters"
+    );
+    return option;
+  }
+
+  //                       STEP 7 :
+  //  in a function declare a list of lowercase , uppercase , numbers and special characters in a string
+  // convert the string into an array and then generate a random
+
+  function randomLowerCaseCharacter() {
+    const lowerCaseCharacters = "abcdefghijklmnopqrstuvwxyz";
+    const lowerCaseArray = lowerCaseCharacters.split("");
+    const generateRandomLowerCC =
+      lowerCaseArray[Math.floor(Math.random() * lowerCaseArray.length)];
+    return generateRandomLowerCC;
+  }
+
+  function randomUppercaseCharacter() {
+    const upperCaseCharacters = "abcdefghijklmnopqrstuvwxyz";
+    const upperCaseArray = upperCaseCharacters.toUpperCase().split("");
+    const generateRandomUpperCC =
+      upperCaseArray[Math.floor(Math.random() * upperCaseArray.length)];
+    return generateRandomUpperCC;
+  }
+
+  function randomNumericalValue() {
+    const numericalValue = "0123456789";
+    const numericalArray = numericalValue.split("");
+    const generateRandomNumericalValue =
+      numericalArray[Math.floor(Math.random() * numericalArray.length)];
+    return generateRandomNumericalValue;
+  }
+
+  function randomSpecialCharacter() {
+    const specialCharacters = "@|!#$%&/()=?»«@£§€{}.-;<>_,";
+    const specialCharactersArray = specialCharacters.split("");
+    const generateRandomSpecialCharacter =
+      specialCharactersArray[
+        Math.floor(Math.random() * specialCharactersArray.length)
+      ];
+    return generateRandomSpecialCharacter;
+  }
+
+  //                       STEP 8  :
+  // store the above functions in a variable
+
+  const passwordLength = userPasswordLength();
+
+  const inputLowerCase = lowerCaseOption();
+
+  const inputUpperCase = upperCaseOption();
+
+  const inputNumericValue = numericOption();
+
+  const inputSpecialCharacter = specialCharacterOption();
+
+  //                       STEP 9  :
+
+  function generatingUserPassword() {
```

Comment:
I don't think there's really a need to separate this section out into its own function. Since you have declared this function inside your main function and then called it straight away inside the same function it is declared in, you could just run the code directly inside the main function. Essentially get rid of line 129, the closing brace on 172 and the return on 173 and it should work the same, with fewer lines.